lib/improver/utilities/solar.py

```diff
@@ -0,0 +1,277 @@
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+""" Utilites to find the relative position of the sun."""
+
+import numpy as np
+import datetime as dt
+
+from improver.utilities.temporal import iris_time_to_datetime
+from improver.utilities.spatial import (
+    lat_lon_determine, transform_grid_to_lat_lon)
+
+
+def solar_declination(day_of_year):
+    """
+    Calculate the Declination for the day of the year.
+
+    Calculation equivalent to the calculation defined in
+    NOAA Earth System Reseach Lab Low Accuracy Equations
+    https://www.esrl.noaa.gov/gmd/grad/solcalc/sollinks.html
+
+    Args:
+        day_of_year (int):
+            Day of the year 0 to 365, 0 = 1st January
+
+    Returns:
+        declination (float):
+            Declination in degrees.North-South
+    """
+    # Declination (degrees):
+    # = -(axial_tilt)*cos(360./orbital_year * day_of_year - solstice_offset)
+    declination = -23.5 * np.cos(np.radians(0.9856 * day_of_year + 9.3))
+    return declination
+
+
+def solar_hour_angle(longitudes, day_of_year, utc_hour):
+    """
+    Calculate the Solar Hour angle for an array of longitudes.
+
+    Calculation equivalent to the calculation defined in
+    NOAA Earth System Reseach Lab Low Accuracy Equations
+    https://www.esrl.noaa.gov/gmd/grad/solcalc/sollinks.html
+
+    Args:
+        longitudes (float or numpy.array):
+            A single Longitude or array of Longitudes
+            longitudes needs to be between 180.0 and -180.0 degrees
+        day_of_year (int):
+            Day of the year 0 to 365, 0 = 1st January
+        utc_hour (float):
+            Hour of the day in UTC
```

Comment:
I think we might want an error for out-of-bounds numbers on day and hour here too.  Since these are external utilities and not part of the plugin, we don't have full defence against invalid inputs.